tensorflow/python/ops/array_ops.py

```diff
@@ -575,9 +579,16 @@ def _slice_helper(tensor, slice_spec, var=None):
     TypeError: If the slice indices aren't int, slice, ellipsis,
       tf.newaxis or scalar int32/int64 tensors.
   """
+  def convert_if_possible(slice_spec):
+    try:
+      return ops.convert_to_tensor(slice_spec)
+    except:
+      pass
+    return None
 
-  if not isinstance(slice_spec, (list, tuple)):
```

Comment:
Note that these deleted lines of code are important for normal indexing: they ensure that cases like `x[i]` have an iterable as `slice_spec`.

Reply:
That is wrong, I will fix it.